tutorial.cpp

```diff
@@ -251,6 +258,91 @@ int TestMatrix() {
+	std::string testFileName = "current/TestMatrixBuilder.out";
+	std::ofstream outFile(testFileName);
+	if (!outFile.is_open()) {
+		std::cout << "FAILURE: Cannot open " << testFileName << "!" << std::endl;
+		return 1;
+	}
+
+	try{
+
+		Matrix mat1(4, 6, 1.);
+		mat1.setName("Matrix 1");
+		Print(mat1, outFile);
+
+		Matrix mat2(4, 2, 2.);
+		mat2.setName("Matrix 2");
+		Print(mat2, outFile);
+
+		Matrix mat3(3, 6, 3.);
+		mat3.setName("Matrix 3");
+		Print(mat3, outFile);
+
+		Matrix combinedMatrix1 = MatrixBuilder::BuildMatrixFromCombination(mat1, mat2, MatrixCombinationType::LeftRight);
+		combinedMatrix1.setName("Combined Matrix 1");
+		Print(combinedMatrix1, outFile);
+
+		Matrix combinedMatrix2 = MatrixBuilder::BuildMatrixFromCombination(mat2, mat1, MatrixCombinationType::LeftRight);
+		combinedMatrix2.setName("Combined Matrix 2");
+		Print(combinedMatrix2, outFile);
+
+		Matrix combinedMatrix3 = MatrixBuilder::BuildMatrixFromCombination(mat1, mat3, MatrixCombinationType::TopDown);
+		combinedMatrix3.setName("Combined Matrix 3");
+		Print(combinedMatrix3, outFile);
+
+		Matrix combinedMatrix4 = MatrixBuilder::BuildMatrixFromCombination(mat3, mat1, MatrixCombinationType::TopDown);
+		combinedMatrix4.setName("Combined Matrix 4");
+		Print(combinedMatrix4, outFile);
+
+		Matrix combinedMatrix5 = MatrixBuilder::BuildMatrixFromCombination(mat2, mat1(0,3,3,4), MatrixCombinationType::TopDown);
+		combinedMatrix5.setName("Combined Matrix 5");
+		Print(combinedMatrix5, outFile);
+
+		Matrix combinedMatrix6 = MatrixBuilder::BuildMatrixFromCombination(mat3, mat2(0,2,1,1), MatrixCombinationType::LeftRight);
+		combinedMatrix6.setName("Combined Matrix 6");
+		Print(combinedMatrix6, outFile);
+
+		Matrix combinedMatrix7 = MatrixBuilder::BuildMatrixFromCombination(mat1(1,3,0,5), mat3, MatrixCombinationType::LeftRight);
+		combinedMatrix7.setName("Combined Matrix 7");
+		Print(combinedMatrix7, outFile);
+
+		Matrix combinedMatrix8 = MatrixBuilder::BuildMatrixFromCombination(mat3(0,2,2,3), mat2, MatrixCombinationType::TopDown);
+		combinedMatrix8.setName("Combined Matrix 8");
+		Print(combinedMatrix8, outFile);
+
+		Matrix combinedMatrix9 = MatrixBuilder::BuildMatrixFromCombination(mat3(0,1,0,5), mat1(2,3,0,5), MatrixCombinationType::LeftRight);
+		combinedMatrix9.setName("Combined Matrix 9");
+		Print(combinedMatrix9, outFile);
+
+		Matrix combinedMatrix10 = MatrixBuilder::BuildMatrixFromCombination(mat1(2,2,2,2), mat2(2,2,0,0), MatrixCombinationType::TopDown);
+		combinedMatrix10.setName("Combined Matrix 10");
+		Print(combinedMatrix10, outFile);
+
+		Matrix combinedMatrix11 = MatrixBuilder::BuildMatrixFromCombination(mat2, mat2, MatrixCombinationType::LeftRight);
+		combinedMatrix11.setName("Combined Matrix 11");
+		Print(combinedMatrix11, outFile);
+
+		Matrix combinedMatrix12 = MatrixBuilder::BuildMatrixFromCombination(mat2, mat2, MatrixCombinationType::TopDown);
+		combinedMatrix12.setName("Combined Matrix 12");
+		Print(combinedMatrix12, outFile);
+

```

Comment:
Add two more classes of tests: (1) matrices whose entries vary as we span the columns and rows and (2) catch the exceptions.

Reply:
Done 1

Reply:
Done 2. Can catch the sizing exceptions. The exceptions for the type of combination don't need to be done because the compiler will not let them through.